h/api/models/test/elastic_test.py

```diff
@@ -7,117 +10,463 @@
+        Document({'dc': {'doi': 'foobar'}}, claimant='http://example.com'),
+        Document({'dc': {'doi': 'doi:foobar'}}, claimant='http://example.com')])
+    def test_uris_generates_doi_uri_from_dc_meta(self, doc):
+        expected = [DocumentURI({'claimant': 'http://example.com',
+                                 'uri': 'doi:foobar',
+                                 'type': 'dc-doi',
+                                 'created': None, 'updated': None}),
+
+                    DocumentURI({'claimant': 'http://example.com',
+                                 'uri': 'http://example.com',
+                                 'type': 'self-claim',
+                                 'created': None, 'updated': None})]
+
+        assert doc.uris.sort() == expected.sort()
+
+
+class TestDocumentMeta(object):
+    def test_created(self):
+        meta = DocumentMeta({'created': datetime.datetime(2016, 2, 25, 16, 45, 23, 371848)})
+        assert meta.created == datetime.datetime(2016, 2, 25, 16, 45, 23, 371848)
+
+    def test_updated(self):
+        meta = DocumentMeta({'updated': datetime.datetime(2016, 2, 25, 16, 45, 23, 371848)})
+        assert meta.updated == datetime.datetime(2016, 2, 25, 16, 45, 23, 371848)
+
+    def test_claimant(self):
+        meta = DocumentMeta({'claimant': 'http://example.com'})
+        assert meta.claimant == 'http://example.com'
+
+    def test_claimant_normalized(self):
+        meta = DocumentMeta({'claimant': 'http://example.com'})
+        assert meta.claimant_normalized == 'http://example.com'
+
+    def test_type(self):
+        meta = DocumentMeta({'type': 'title'})
+        assert meta.type == 'title'
+
+    def test_type_normalizes_multiple_dots(self):
+        meta = DocumentMeta({'type': 'dc..description'})
+        assert meta.type == 'dc.description'
+
+    def test_type_normalizes_case(self):
+        meta = DocumentMeta({'type': 'dc.Contributor.Sponsor'})
+        assert meta.type == 'dc.contributor.sponsor'
+
+    def test_type_normalizes_colons(self):
+        meta = DocumentMeta({'type': 'facebook.book:isbn'})
+        assert meta.type == 'facebook.book.isbn'
+
+    def test_value(self):
+        meta = DocumentMeta({'value': 'Example Page'})
+        assert meta.value == 'Example Page'
+
+
+class TestDocumentURI(object):
+    def test_created(self):
+        docuri = DocumentURI({'created': datetime.datetime(2016, 2, 25, 16, 45, 23, 371848)})
+        assert docuri.created == datetime.datetime(2016, 2, 25, 16, 45, 23, 371848)
+
+    def test_updated(self):
+        docuri = DocumentURI({'updated': datetime.datetime(2016, 2, 25, 16, 45, 23, 371848)})
+        assert docuri.updated == datetime.datetime(2016, 2, 25, 16, 45, 23, 371848)
+
+    def test_claimant(self):
+        docuri = DocumentURI({'claimant': 'http://example.com'})
+        assert docuri.claimant == 'http://example.com'
+
+    def test_claimant_normalized(self):
+        docuri = DocumentURI({'claimant': 'http://example.com'})
```

Comment:
Want to check that it actually gets normalised? (Don't bother mocking out the normaliser... just stick a trailing slash on it or something.)